app/Http/Requests/Settings/ChangeLoginRequest.php

```diff
@@ -2,96 +2,96 @@
-	 * {@link \App\Actions\Settings\Login::do()} enforces correct
-	 * authorization.
-	 *
-	 * @return bool always true
 	 */
 	public function authorize(): bool
 	{
-		/*return AccessControl::is_logged_in() && (
-			AccessControl::is_admin() ||
-			!AccessControl::user()->is_locked
-		);*/
-
-		return true;
+		return Gate::check(UserPolicy::EDIT_SETTINGS, User::class);
 	}
 
 	/**
 	 * {@inheritDoc}
 	 */
 	public function rules(): array
 	{
-		$rules = parent::rules();
-		$rules[self::OLD_USERNAME_ATTRIBUTE] = ['sometimes', new UsernameRule()];
-		$rules[self::OLD_PASSWORD_ATTRIBUTE] = ['sometimes', new PasswordRule(false)];
-
-		return $rules;
+		return [
+			HasUsername::USERNAME_ATTRIBUTE => ['sometimes', new UsernameRule()],
+			HasPassword::PASSWORD_ATTRIBUTE => ['required', new PasswordRule(false)],
+			self::OLD_PASSWORD_ATTRIBUTE => ['required', new PasswordRule(false)],
+		];
 	}
 
 	/**
 	 * {@inheritDoc}
 	 */
 	protected function processValidatedValues(array $values, array $files): void
 	{
-		parent::processValidatedValues($values, $files);
-		$this->oldUsername = $values[self::OLD_USERNAME_ATTRIBUTE] ?? null;
-		if (array_key_exists(self::OLD_PASSWORD_ATTRIBUTE, $values)) {
-			// See {@link HasPasswordTrait::password()} for an explanation
-			// of the semantic difference between `null` and `''`.
-			$this->oldPassword = $values[self::OLD_PASSWORD_ATTRIBUTE] ?? '';
+		$this->password = $values[HasPassword::PASSWORD_ATTRIBUTE];
+		$this->oldPassword = $values[self::OLD_PASSWORD_ATTRIBUTE];
+
+		// We do not allow '' as a username. So any such input will be cast to null
+		if (array_key_exists(HasUsername::USERNAME_ATTRIBUTE, $values)) {
+			$this->username = trim($values[HasUsername::USERNAME_ATTRIBUTE]);
+			$this->username = $this->username === '' ? null : $this->username;
 		} else {
-			$this->oldPassword = null;
+			$this->username = null;
 		}
 	}
 
 	/**
+	 * Returns the previous password.
+	 *
+	 * See {@link HasPasswordTrait::password()} for an explanation of the
+	 * semantic difference between the return values `null` and `''`.
+	 *
 	 * @return string|null
 	 */
-	public function oldUsername(): ?string
+	public function oldPassword(): ?string
+	{
+		return $this->oldPassword;
+	}
+
+	/**
+	 * Return the new username chosen.
+	 * if Username is null, this means that the user does not want to update it.
+	 *
+	 * @return ?string
+	 */
+	public function username(): ?string
 	{
-		return $this->oldUsername;
+		return $this->username;
 	}
 
 	/**
-	 * Returns the previous (old) password, if available.
+	 * Handle a failed authorization attempt.
 	 *
-	 * See {@link HasPasswordTrait::password()} for an explanation of the
-	 * semantic difference between the return values `null` and `''`.
+	 * @return void
 	 *
-	 * @return string|null
+	 * @throws \Illuminate\Auth\Access\AuthorizationException
 	 */
-	public function oldPassword(): ?string
+	protected function failedAuthorization(): void
```

Comment:
Why is this method overwritten? What is wrong with the base method? Can it be removed?

Reply:
removed.